### ISSUE TITLE
Support for inline images

### ISSUE THREAD

**Issue by freb** [baliance/gooxml#134]
I need to insert an inline image. I see convenience functions for anchored images but not for inline images.

I took a stab at adapting the `AddDrawingAnchored` method on `Run`, but the image is showing blank.

My stab:

~~~go
func AddDrawingInline(r document.Run, img common.ImageRef) error {
	ic := wml.NewEG_RunInnerContent()
	r.X().EG_RunInnerContent = append(r.X().EG_RunInnerContent, ic)

	ic.Drawing = wml.NewCT_Drawing()

	inline := wml.NewWdInline()
	ic.Drawing.Inline = append(ic.Drawing.Inline, inline)

	inline.Graphic = dml.NewGraphic()
	inline.Graphic.GraphicData = dml.NewCT_GraphicalObjectData()
	inline.Graphic.GraphicData.UriAttr = "http://schemas.openxmlformats.org/drawingml/2006/picture"
	inline.Extent.CxAttr = int64(float64(img.Size().X*measurement.Pixel72) / measurement.EMU)
	inline.Extent.CyAttr = int64(float64(img.Size().Y*measurement.Pixel72) / measurement.EMU)

	randID := 0x7FFFFFFF & rand.Uint32()
	inline.DocPr.IdAttr = randID
	p := pic.NewPic()
	p.NvPicPr.CNvPr.IdAttr = randID

	// find the reference to the actual image file in the document relationships
	// so we can embed via the relationship ID
	imgID := img.RelID()
	if imgID == "" {
		return errors.New("couldn't find reference to image within document relations")
	}

	inline.Graphic.GraphicData.Any = append(inline.Graphic.GraphicData.Any, p)

	return nil
}

~~~

Is this something you would consider adding? For my current use case, I'm trying to rely on the paragraph position and formatting to handle image location and an inline image seems to work best from inside Word.

**Comment by tbaliance** [comment]
There's a new ```AddDrawingInline``` method on ```Run``` that worked in my testing.  I'll look at fleshing out the InlineDrawing later, but for now it's good enough to insert and allow you to control the image size.

**Comment by freb** [comment]
This definitely works. But I noticed I get an error when I try to open it in Word: `We're sorry. We can't open out.docx because we found a problem with its contents`. Once I click the button to recover it, everything looks fine.

I diffed the `document.xml` between the pre and post recovered versions and there are some differences, but nothing that looks wrong necessarily. I tried replacing each different element in turn manually and never got the message to go away.

I noticed that I was getting the same thing with the anchored images before but I just assumed I wasn't setting all the required properties.

When adding an inline image on a blank template or a `document.New()`, I don't get any errors.

Here is the code I am using after adding just a few title elements (Note that I am `document.Open("template.docx")` which is a file that includes a header and footer on the first page and some styles):

~~~
img, err := common.ImageFromFile("logo_client.png")
iref, err := doc.AddImage(img)
if err != nil {
	log.Fatalf("unable to add image to document: %v", err)
}

para := doc.AddParagraph()
para.Properties().SetSpacing(42*measurement.Point, 0*measurement.Point)
para.Properties().SetAlignment(wml.ST_JcCenter)
inline, err := para.AddRun().AddDrawingInline(iref) // inline, err :=
if err != nil {
	log.Fatalf("error inserting inline image %v: %v", inline, err)
}
~~~

Like I said, the document recovers fine so its not a big issue, but thought it might be worth reporting.

**Comment by tbaliance** [comment]
That means we’re doing something wrong. I tested on OSX Word which is usually the most particular regarding format. What version of word reports an error?

**Comment by freb** [comment]
Microsoft Office 365, Version 1710 (Build 8625.2139), on Windows. Should be the most up-to-date version of 365.

I just reproduced the "can't open" issue with a brand new blank word document. For some reason, when I create a new word document the default is "compatibility" mode, but I get the same result whether I leave it in compatibility mode or upgrade it to the "newest file format". It doesn't give any indication as to what the actual format is before or after, but its probably in the .xml somewhere.

[out-before-recover.docx](https://github.com/baliance/gooxml/files/1549277/out-before-recover.docx)
[out-after-recover.docx](https://github.com/baliance/gooxml/files/1549279/out-after-recover.docx)
[test.png](https://user-images.githubusercontent.com/1723508/33852275-b7c832e4-de6e-11e7-8cfa-bc0872d1c551.png)

I uploaded the before and after recovery files (these show compatibility mode when open in Word, I did not upgrade them). I also attached the image I'm using and here is the code I ran:

~~~
package main

import (
	"log"

	"baliance.com/gooxml/document"
	"baliance.com/gooxml/common"
)

func main() {
	doc, err := document.Open("testimg.docx")
	if err != nil {
		log.Fatalf("error opening Windows Word 2016 document: %s", err)
	}

	img, err := common.ImageFromFile("test.png")
	iref, err := doc.AddImage(img)
	if err != nil {
		log.Fatalf("unable to add image to document: %v", err)
	}

	para := doc.AddParagraph()
	inline, err := para.AddRun().AddDrawingInline(iref)
	if err != nil {
		log.Fatalf("error inserting inline image %v: %v", inline, err)
	}

	doc.SaveToFile("out.docx")
	return
}
~~~
